src/tests/java/htsjdk/samtools/SAMRecordUnitTest.java

```diff
@@ -94,4 +95,214 @@ public void testOffsetAtRead(String cigar, int expectedReferencePos, int posInRe
             SAMRecord sam = new SAMRecordSetBuilder().addFrag("test", 0, 1, false, false, cigar, null, 2);
             Assert.assertEquals(sam.getReferencePositionAtReadPosition(posInRead), expectedReferencePos);
     }
+
+    @DataProvider(name = "deepCopyTestData")
+    public Object [][] deepCopyTestData() {
+        return new Object[][]{
+                {"3S9M"},
+                {"4M1I6M"},
+        };
+    }
+
+    @Test(dataProvider = "deepCopyTestData")
+    public void testDeepCopyRef(String cigar) {
+        SAMRecord sam = new SAMRecordSetBuilder().addFrag("test", 0, 1, false, false, cigar, null, 2);
```

Comment:
Why not just have the `deepCopyTestData` `DataProvider` build and return these `SAMRecords`, since it appears that each test is creating them in identical fashion from the `Cigar`s